src/extensions/default/AutoUpdate/main.js

```diff
@@ -0,0 +1,833 @@
+     */
+    function setupAutoUpdatePreference(){
+        PreferencesManager.definePreference("autoUpdate.AutoUpdate", "boolean", true, {
+            description: Strings.DESCRIPTION_AUTO_UPDATE
+        });
+
+        // Set or unset the auto update, based on preference state change
+        PreferencesManager.on("change", "autoUpdate.AutoUpdate", function () {
+            if(_isAutoUpdateEnabled()) {
+                setupAutoUpdate();
+                UpdateNotification.registerUpdateHandler(_updateProcessHandler);
+            } else {
+                unsetAutoUpdate();
+                UpdateNotification.registerUpdateHandler(null);
+            }
+        });
+    }
+
+
+    /**
+     * Checks if auto update preference is enabled or disabled
+     * @private
+     * @returns {boolean} - true if preference enabled, false otherwise
+     */
+    function _isAutoUpdateEnabled() {
+        return (PreferencesManager.get("autoUpdate.AutoUpdate") !== false);
+    }
+    
+    
+    /**
+     * Initializes the state of parsed content from updateHelper.json
+     */
+    function initState() {
+        updateJsonHandler.parse()
+            .done(function () {
+                checkUpdateStatus();
+            })
+            .fail(function (code) {
+                var logMsg;
+                switch (code) {
+                case StateHandlerMessages.FILE_NOT_FOUND:
+                    logMsg = "AutoUpdate : updateHelper.json cannot be parsed, does not exist";
+                    break;
+                case StateHandlerMessages.FILE_NOT_READ:
+                    logMsg = "AutoUpdate : updateHelper.json could not be read";
+                    break;
+                case StateHandlerMessages.FILE_PARSE_EXCEPTION:
+                    logMsg = "AutoUpdate : updateHelper.json could not be parsed, exception encountered";
+                    break;
+                case StateHandlerMessages.FILE_READ_FAIL:
+                    logMsg = "AutoUpdate : updateHelper.json could not be parsed";
+                    break;
+                }
+                console.log(logMsg);
+            });
+    }
+
+    /**
+     * Sets up the Auto Update environment
+     */
+    function setupAutoUpdate() {
+        updateJsonHandler = new StateHandler(updateJsonPath);
+
+        updateDomain.exec('initNode', {
+            messageIds: MessageIds,
+            updateDir: updateDir
+        });
+
+        updateDomain.on('data', receiveMessageFromNode);
+        initState();
+    }
+
+    
+    /**
+     * Unsets the Auto Update environment
+     */
+    function unsetAutoUpdate() {
+        updateJsonHandler = null;
+        updateDomain.off('data');
+        resetAppQuitHandler();
+    }
+    
+    /**
+     * Overriding the appReady for Auto update
+     */
+
+    AppInit.appReady(function () {
+        
+        // Auto Update is supported on Win and Mac, as of now
+        if (brackets.platform !== "linux") {
```

Comment:
I think one of the review comments got missed here. You need to check if `brackets.app.setUpdateParams` is registered or not. Only then run through the flow.